R/add_dd.R

```diff
@@ -0,0 +1,183 @@
+#' Add Delay Discharge to working file
+#'
+#' @param data The input data frame
+#' @param year The year being processed
+#'
+#' @return A data frame linking delay discharge cohorts
+#' @export
+#'
+#' @family episode file
+add_dd <- function(data, year) {
+  year_param <- year
+
+  data <- data %>%
+    dplyr::arrange(chi, cij_marker) %>%
+    dplyr::mutate(
+      cij_start_date_lower = cij_start_date - lubridate::days(1),
+      cij_end_date_upper = cij_end_date + lubridate::days(1)
+    )
+
+  ## handling DD ----
+  dd_data <- read_file(get_source_extract_path(year_param, "DD"))
+  by_dd <- dplyr::join_by(
+    chi,
+    x$keydate1_dateformat >= y$cij_start_date_lower,
+    x$keydate2_dateformat <= y$cij_end_date_upper
+  )
+  data <- dd_data %>%
+    dplyr::inner_join(data,
+      by_dd,
+      suffix = c("_dd", "")
+    ) %>%
+    dplyr::arrange(cij_start_date, cij_end_date, cij_marker, postcode) %>%
+    # remove duplicate columns
+    dplyr::distinct(
+      cij_start_date,
```

Comment:
I think CHI needs to be included here.
```suggestion
    dplyr::distinct(
      chi,
      cij_start_date,
```